ios/RNNetPrinter.m

```diff
@@ -1,190 +1,72 @@
-}
-
-RCT_EXPORT_METHOD(connectPrinter:(NSString *)host
-                  withPort:(nonnull NSNumber *)port
-                  success:(RCTResponseSenderBlock)successCallback
-                  fail:(RCTResponseSenderBlock)errorCallback) {
-    @try {
-        BOOL isConnectSuccess = [[PrinterSDK defaultPrinterSDK] connectIP:host];
-        !isConnectSuccess ? [NSException raise:@"Invalid connection" format:@"Can't connect to printer %@", host] : nil;
-
-        connected_ip = host;
-        [[NSNotificationCenter defaultCenter] postNotificationName:@"NetPrinterConnected" object:nil];
-        successCallback(@[[NSString stringWithFormat:@"Connecting to printer %@", host]]);
-
-    } @catch (NSException *exception) {
-        errorCallback(@[exception.reason]);
-    }
-}
 
+RCT_EXPORT_MODULE()
 
 RCT_EXPORT_METHOD(connectAndSend:(NSString *)host
                   withPort:(nonnull NSNumber *)port
                   printRawData:(NSString *)text
                   success:(RCTResponseSenderBlock)successCallback
                   fail:(RCTResponseSenderBlock)errorCallback) {
     @try {
-        BOOL isConnectSuccess = [[PrinterSDK defaultPrinterSDK] connectIP:host];
-        !isConnectSuccess ? [NSException raise:@"Invalid connection" format:@"Can't connect to printer %@", host] : nil;
-        [[PrinterSDK defaultPrinterSDK] sendHex:text];
-        successCallback(@[[NSString stringWithFormat:@"Sent to printer %@", host]]);
-
+        int result = EPOS2_SUCCESS;
+        printer_ = [[Epos2Printer alloc] initWithPrinterSeries:1 lang:EPOS2_MODEL_ANK];
+        [printer_ setReceiveEventDelegate:self];
+
+        NSString* target = [NSString stringWithFormat:@"TCP:%@", host];
+        result = [printer_ connect:target timeout:5000];
+        if (result != EPOS2_SUCCESS) {
+            [NSException raise:@"Invalid connection" format:@"Can't connect to printer %@", host];
+        }
+        NSData* payload = [NSData dataWithBase64EncodedString:text];
+        [printer_ addCommand:payload];
+        result = [printer_ sendData:5000];
+        if (result != EPOS2_SUCCESS) {
+            [NSException raise:@"Print failed" format:@"Error occurred while printing"];
+        }
+        _successCallback = successCallback;
+        _errorCallback = errorCallback;
+        [printer_ clearCommandBuffer];
+        [printer_ disconnect];
     } @catch (NSException *exception) {
         errorCallback(@[exception.reason]);
     }
 }
 
-RCT_EXPORT_METHOD(printRawData:(NSString *)text
-                  fail:(RCTResponseSenderBlock)errorCallback) {
-    @try {
-
-        !connected_ip ? [NSException raise:@"Invalid connection" format:@"Can't connect to printer"] : nil;
-
-        [[PrinterSDK defaultPrinterSDK] sendHex:text];
-    } @catch (NSException *exception) {
-        errorCallback(@[exception.reason]);
+- (void) onPtrReceive:(Epos2Printer *)printerObj code:(int)code status:(Epos2PrinterStatusInfo *)status printJobId:(NSString *)printJobId
```

Comment:
where is this `onPtrReceive` invoked from?

Reply:
We attach this function to printer object above.